workflow/rules/long_read.smk

```diff
@@ -119,40 +97,50 @@ rule canu_correct:
         "../envs/canu.yaml"
     threads: 16
     shell:
-        "( if [ -d {params.outdir} ]; then rm -Rf {params.outdir}; fi &&"
-        " canu -correct -nanopore {input} -p {wildcards.sample} -d {params.outdir} genomeSize=30k minOverlapLength=10 minReadLength=200"
-        " useGrid=false {params.for_testing}"
-        " corMMapMerSize=10 corOutCoverage=50000 corMinCoverage=0 maxInputCoverage=20000"
-        " corOverlapper=minimap utgOverlapper=minimap obtOverlapper=minimap"
-        " corConcurrency={params.concurrency}"
-        " cormhapConcurrency={params.concurrency} cormhapThreads={params.concurrency}"
-        " cormmapConcurrency={params.concurrency} cormmapThreads={params.concurrency}"
-        " obtmmapConcurrency={params.concurrency} obtmmapThreads={params.concurrency}"
-        " utgmmapConcurrency={params.concurrency} utgmmapThreads={params.concurrency}"
-        " redConcurrency={params.concurrency} redThreads={params.concurrency}"
-        " ovbConcurrency={params.concurrency}"
-        " ovsConcurrency={params.concurrency}"
-        " oeaConcurrency={params.concurrency})"
-        "> {log} 2>&1"
-
-
-# rule medaka_consensus_reference:
-use rule assembly_polishing_ont as medaka_consensus_reference with:
+        """
+        ( if [ -d {params.outdir} ]; then rm -Rf {params.outdir}; fi &&
+        canu -correct -nanopore {input} -p {wildcards.sample} -d {params.outdir} genomeSize=30k minOverlapLength=10 minReadLength=200 \
+        useGrid=false {params.for_testing} \
+        corMMapMerSize=10 corOutCoverage=50000 corMinCoverage=0 maxInputCoverage=20000 \
+        corOverlapper=minimap utgOverlapper=minimap obtOverlapper=minimap \
+        corConcurrency={params.concurrency} \
+        cormhapConcurrency={params.concurrency} cormhapThreads={params.concurrency} \
+        cormmapConcurrency={params.concurrency} cormmapThreads={params.concurrency} \
+        obtmmapConcurrency={params.concurrency} obtmmapThreads={params.concurrency} \
+        utgmmapConcurrency={params.concurrency} utgmmapThreads={params.concurrency} \
+        redConcurrency={params.concurrency} redThreads={params.concurrency} \
+        ovbConcurrency={params.concurrency} \
+        ovsConcurrency={params.concurrency} \
+        oeaConcurrency={params.concurrency})
+        gzip -d {output.corr_gz}
+        2> {log}
+        """
+
+
+rule clip_adbc_corrected:
     input:
-        fasta="results/{date}/corrected/{sample}/{sample}.correctedReads.fasta.gz",
-        reference="resources/genomes/main.fasta",
+        primer=get_artic_primer,
+        reads="results/{date}/corrected/{sample}/{sample}.correctedReads.fasta",
+        ref_genome="resources/genomes/main.fasta",
     output:
-        temp("results/{date}/consensus/medaka/{sample}/{sample}.fasta"),
+        "results/{date}/norm_trim_corr_reads/{sample}/{sample}.clip.fasta",
```

Comment:
Just following up on the "no using corrected reads as input for medaka":

The rule `assembly_polishing_ont` also has them as input, so I decided to exchange it for the clipped reads. Should the output from clip_adbc_corrected `norm_trim_corr_reads/{sample}/{sample}.clip.fasta` or the output from downsample_and_trim_raw `results/{date}/norm_trim_raw_reads/{sample}/{sample}.cap.clip.fasta` should be used for that?

Moreover, `results/{date}/norm_trim_corr_reads/{sample}/{sample}.clip.fast` isn't used right now.

Reply:
For polishing always the raw reads should be used, because during read correction information is lost, which actually the polishing algorithm is supposed to use to decide if this information represents real variants or only noise. Read correction denoises, but only based on the information contained in the reads, whereas the polishing algorithm can relate to the reference. So if we do read correction (which is intended to simplify [de-novo] assembly and not to find variants) we actually would deprive the polisher of crucial information it is specialized to use together with the reference to identify variants. It will still work for clear cases of course, but it would be methodically wrong and introduce a bias.
Therefore:      
results/{date}/norm_trim_raw_reads/{sample}/{sample}.cap.clip.fasta 
is the rigth file to use